src/Servers/IIS/IIS/src/Core/IISServerAuthenticationHandler.cs

```diff
@@ -12,7 +12,7 @@ namespace Microsoft.AspNetCore.Server.IIS.Core
     /// <summary>
     /// The default authentication handler with IIS In-Process
     /// </summary>
-    [Obsolete("The IISServerAuthenticationHandler is obsolete and will be removed in a future release.")]
+    [Obsolete("The IISServerAuthenticationHandler is obsolete and may be removed in a future release.")] // Never remove.
```

Comment:
@jkotalik. This is one of the ones that I'm most unsure about. Do you see this being painful to maintain and warranting removal after .NET 6? Personally, I don't see much harm in leaving an obsolete type in the `Microsoft.AspNetCore.Server.IIS.Core` namespace, and it's nice not to break people.

Reply:
I'd argue that these types were never intended to be public and shouldn't be used. I don't see why we should never remove these? 

FYI we made these obsolete in .NET 6.

Reply:
I am not sure about IISServerAuthenticationHandler, but I don't see why we'd ever remove something like WriteOnlyStream which costs practically nothing to maintain, lives in [a namespace whose only public types are obsolete](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.server.iis.core?view=aspnetcore-5.0), and whose removal could potentially break old libraries that depend those types.

Reply:
We already scanned for any libraries that depend on these types, and the number of hits were in the single digits across all of github and nuget.

This isn't a hill I'm willing to die on, I just think the harm of deleting these types is very low and would clean up an entire namespace that shouldn't have been public.